Switch to go-modules

saschagrunert opened the pull request:
This removes the old vendoring for go module support. We also add an
additional verification step to travis. Documentation has been updated
as well.

---
After that, I would like to update to the latest Kubernetes alpha to propose the new CRI API changes which are introduced with dual stack IPv6 support, WDYT?

feiskyer commented:
The idea looks good, but I wonder why so many vendor files are removed and updated? Generally, if we only switch go modules, the vendor files should still be same, right?

feiskyer commented:
/cc @yujuhong @Random-Liu

saschagrunert commented:
> The idea looks good, but I wonder why so many vendor files are removed and updated? Generally, if we only switch go modules, the vendor files should still be same, right?

Ah I'm sorry, I also updated the dependencies as well. The thing is that dependent projects switched to go modules too, which will lead into ma much cleaner go.mod file.

feiskyer commented:
@saschagrunert Thanks, then it makes sense.

feiskyer commented on hack/tree_status.sh:
```diff
@@ -14,17 +14,14 @@
-sort vendor.conf -o vendor.conf
-
-echo "Please commit the change made by this file..."
+STATUS=$(git status --porcelain)
```
I think we should also run go mod command before checking git status, e.g. 
```
$(GO) mod tidy && \
		$(GO) mod vendor && \
		$(GO) mod verify
```

saschagrunert replied:
Okay, added this as well.

feiskyer commented on hack/tree_status.sh:
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
```
could we rename this file to hack/verify-go-modules.sh?

saschagrunert replied:
Yes, changed it.

feiskyer commented on go.mod:
```diff
@@ -0,0 +1,53 @@
+	k8s.io/apimachinery v0.0.0
+	k8s.io/client-go v11.0.0+incompatible
+	k8s.io/cri-api v0.0.0
+	k8s.io/kubernetes v0.0.0
```
v0.0.0 means latest. For k8s.io dependencies, do we need to change this to a specific version, e.g. v1.15.0?

saschagrunert replied:
Not necessarily since we override it in the `replace` section: `k8s.io/kubernetes => k8s.io/kubernetes v1.15.1`

feiskyer commented:
/lgtm
/approve
